File: llm_generated_code/Gemini/CPP/Level1/Q4.cpp
#include <iostream>
#include <vector>

using namespace std;

int maxProfit(vector<int>& prices) {
    int n = prices.size();
    if (n <= 1) {
        return 0;
    }

    int minPrice = prices[0];
    int maxProfit = 0;

    for (int i = 1; i < n; ++i) {
        minPrice = min(minPrice, prices[i]);
        maxProfit = max(maxProfit, prices[i] - minPrice);
    }

    return maxProfit;
}

int main() {
    vector<int> prices = {7, 1, 5, 3, 6, 4};
    int maxProfit = maxProfit(prices);
    cout << "Maximum profit: " << maxProfit << endl;

    return 0;
}

File: llm_generated_code/Gemini/Python/Level1/Q4.py
def max_profit(prices):
  """
  Finds the maximum profit achievable by buying and selling a stock once.

  Args:
      prices: A list of stock prices on each day.

  Returns:
      The maximum profit achievable.
  """

  if not prices:
    return 0

  min_price = prices[0]  # Initialize minimum price to first day's price
  max_profit = 0  # Initialize maximum profit to 0

  for price in prices:
    # Update minimum price if current price is lower
    min_price = min(min_price, price)
    # Calculate potential profit for each day by subtracting min price
    profit = price - min_price
    # Update maximum profit if current profit is higher
    max_profit = max(max_profit, profit)

  return max_profit
